hal/inc/ble_hal.h

```diff
@@ -321,9 +321,12 @@ typedef struct hal_ble_pairing_status_updated_evt_t {
 
 typedef struct hal_ble_pairing_auth_data_t {
     hal_ble_pairing_auth_data_type_t type;
+    uint16_t size; // payload length of params
```

Comment:
Let's rename this to `params_size` or something like that. It's very easy to confuse with the length of `hal_ble_pairing_auth_data_t`, which I was actually initially suggesting for extensibility in case we ever have to add something else to this struct :)

Reply:
In that case, we'd better add the version field as well. Because the new added member is not present in old user application and this member is not initialized, which will lead to unknown behavior.

Reply:
We still have the type field which has somewhat a similar purpose.

Reply:
Just in case that the new added member is not type related.